src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -323,16 +327,16 @@ export function TransactionRow ({ transaction, setActionsVisibleId, actionsVisib
     fullNotes = isOffer ? ` Declined Offer: ${notes}` : ` Declined Request: ${notes}`
   } else fullNotes = notes
 
-  const disabledTransaction = disableActionedTransaction(transaction.id)
-  let highlightGreen, highlightRed, highlightNeutral
-  if (disabledTransaction) {
-    highlightGreen = increaseAction()
-    highlightRed = decreseAction()
-    highlightNeutral = statusQuoAction()
-  }
+  const findTransactionById = (transactionId, arrayOfTransactions) => arrayOfTransactions.find(tx => tx.id === transactionId)
+
+  // NB: once modalActionedTransaction no longer has an array or transactions, we can update to:  disabledTransaction = transaction.id === modalActionedTransaction.transaction.id
+  const disabledTransaction = findTransactionById(transaction.id, modalActionedTransaction)
```

Comment:
so this is cheating :)

The idea behind the refactor was that the TransactionRow itself is responsible for storing the state it needs to know how to display. Here you're passing that in through the `confirmationModalProperties`.

`ConfimationModal` should be the only consumer of `confirmationModalProperties`